src/config/builder.rs

```diff
@@ -452,10 +455,14 @@ mod tests {
     /// serializes, or the implementation of `serde_json` has changed. If this test fails, we
     /// should ideally be able to fix so that the original hash passes!
     fn version_hash_match() {
-        assert_eq!(
-            "bc0825487e137ee1d1fc76c616795d041c4825b4ca5a7236455ea4515238885c",
-            ConfigBuilder::default().sha256_hash()
-        );
+        let expected_hash = "6c98bea9d9e2f3133e2d39ba04592d17f96340a9bc4c8d697b09f5af388a76bd";
```

Comment:
Given these changes, this hash will be different on every version bump. We should update the doc comment above accordingly:
- a hash changing can mean the version was bumped
- we should not expect to fix the test to get the original hash

Alternatively, I think we could remove this test and keep only the new lines below (i.e. test that changing version changes the hash).

Reply:
> we should not expect to fix the test to get the original hash

I don't really agree with that. We should keep ensuring that for a given version and a given configuration, the hash should remain the same 🤔 Therefore, fixing the version for that test should be a good way to ensure the feature remains unbroken.

Reply:
Ah, yes that's fair. Apologies, I misread some of the code here.